### ISSUE TITLE
Is there a plan for i18n?

### ISSUE THREAD

**Issue by htve** [ForkIssues/TrackerWin#392]
I am a user from China. Although English is an international language, I prefer to see my mother tongue. I believe that most users have such a demand, WDYT?

**Comment by felixhirt** [comment]
Maybe the string resources could be provided in .xliff format in a separate repository so that users could provide localizations?

**Comment by Otiel** [comment]
Or use a service such as [Weblate](https://weblate.org).

**Comment by DanPristupov** [comment]
[just removed the message accidentally, so this is the second attempt]

A small step back. Git is very complicated. It takes quite a lot of time to understand how it works and not be afraid of losing some data occasionally.

Git names are already confusing. However it works quite simple now. When somebody doesn't know what is 'X', he/she can google it and find a good explanation (even a localized one!).

Translating UI to a different language will bring us to a brand new level of confusion (`sqrt(2)`).

Just think about googling the following cases if you translate them to your native language:

* checkout branch
* cherry-pick
* HEAD
* fast-forward
* reset
* squash/fixup

Rephrasing the famous quote: "Before, only our senior developer and google knew what 'fast-forward merge' means, after translation even they don't know".

As the result I think harder learning curve is better than the way full of guessing and confusion.
Another big plus, is that when you switch to other git client (of from other one, hopefully) you will feel at home. 

TLDR: No.

**Comment by felixhirt** [comment]
Thats a good point, i think SourceTree is translated and if i assist colleagues which use the german version i have no clue which button does what. I guess to use git you should understand english anyway

**Comment by htve** [comment]
Even with the localization of the UI, you can still change to the English UI by yourself. Some people prefer the native language, while others have legal requirements, such as French.

**Comment by jerone** [comment]
_I thought this was asked before, but can't [find](https://github.com/ForkIssues/TrackerWin/search?q=translate+is%3Aissue&type=Issues) the issue._

[GIT Core is translated](https://github.com/git/git/tree/master/po) in multiple [languages](https://github.com/git-l10n/git-po/#readme).

There is also an [effort to translate](https://github.com/jnavila/git-manpages-l10n) GIT [documentation](https://git-scm.com/doc).
For example; [`git-init`](https://git-scm.com/docs/git-init) is already translated in [English](https://git-scm.com/docs/git-init/en) and [French](https://git-scm.com/docs/git-init/fr).

This shows that there is more then enough interest to get GIT translated, and hopefully GIT GUI's (like Fork) also.

**Comment by felixhirt** [comment]
you are right, but two things to note:
- The commands themselves are not translated, only docs and error messages etc.
- As a likely consequence, at least in the german translations i quickly browsed, they use the english words for all the commands and terminology (commit, branch, HEAD, pull, push, fetch, stash, merge etc.)
So if it would be translated, the buttons etc. should still use the english words i guess (this would also avoid the confusion you get from german sourcetree, where they tried to translate the command names...)

**Comment by shyn** [comment]
coz of no Chinese version. Our team has chosen SourceTree recently. Personally, I prefer Fork.

> Git names are already confusing. However it works quite simple now. When somebody doesn't know what is 'X', he/she can 
> google it and find a good explanation (even a localized one!).

This is not an issue as long as we practiced to understand it. People will finally figure it out with their native language.

Translation can be outsourced to volunteers just like SourceTree did.

**Comment by shelllee** [comment]
> 

However, there is a real truth that cannot be ignored. In some product teams in a non-native English speaking country, the English ability of developers is up to the standard while other not, but the proportion of developers is less than 20% in the team, So localization is still necessary.

Moreover, for non code parts in a project, the difference between using fast forward command or not is not so important and so are many other commands. So people may not require all commands to be localized. Maybe, it is enough to deal with only 20% or even little of all git commands. The UI will looks strange, but we are using chrome to read papers in other language, using some software which contains several plugins developed in other language, using ... That's the truth, isn't it?

The only needed is to provide a way to config the UI, and our users will find the way they out.

One more thing, if someone whoes English is poor cannot understand git commands in their native language, could they understand it in English?

**Comment by holysheng** [comment]
软件是为人服务的，而不是人为软件服务的。
我只是觉的这个软件好，但要建立在我用的舒服的前提下。
如果没有中文的UI支持，我也只能先选择其它软件了，况且我的选择有很多，市面上支持中文的Git软件也很多。
因为我是中国人，我喜欢中文。
感谢。

**Comment by DaveBoy** [comment]
The need for translation is like the choice from the command line to the GUI, the people who use the command line do not understand the people who use the GUI, and they will wonder why a simple command or IDE comes with it, why use specialized GUI software?
The answer is the same, we can use the command line, but we prefer to just click and avoid the cumbersome input, which is not a solution to some difficulties, but a more convenient option.
The same is true for translation problems, it's not that we can't understand English or can't google to translate words that we don't understand, we just prefer to see the words we are familiar with, which is also a more convenient option (when you encounter a command that you can't remember, go to google to compare and use the gui to click, just like when you encounter a word you don't understand, go to google to compare and directly watch familiar text)

**Comment by dingshaohua-com** [comment]
the work is too hard? why not resove it?

**Comment by shelllee** [comment]
It's the year 6202, and I'm using LLM agents to submit. How about you guys?
